src/s1reader/s1_burst_slc.py

```diff
@@ -617,6 +623,313 @@ def az_carrier_components(self, offset, position):
+                                                        1,
+                                                        gdal.GDT_Float64,
+                                                        'ENVI')
+                                              for filename_llh in list_filename_llh]
+
+        rdr2geo_obj.topo(dem_raster, lon_raster, lat_raster, hgt_raster)
+
+        # make sure that the ISCE3 rasters are written out to file system
+        lat_raster.close_dataset()
+        lon_raster.close_dataset()
+        hgt_raster.close_dataset()
+
+        # Do the bunch of computation
+        kappa_steer_vec = (2 * (np.linalg.norm(vec_vel_intp, axis=1))
+                             / isce3.core.speed_of_light * self.radar_center_frequency
+                             * self.azimuth_steer_rate)
+
+        kappa_steer_grid = (kappa_steer_vec[..., np.newaxis]
+                            * np.ones(grid_tau.shape[1])[np.newaxis, ...])
+
+        t_burst = (grid_t[0, 0] + grid_t[-1, 0]) / 2.0
+        index_mid_burst_t = int(grid_t.shape[0] / 2 + 0.5)
+        tau_burst = (grid_tau[index_mid_burst_t, 0] 
+                     + grid_tau[index_mid_burst_t, -1]) / 2.0
+        tau0_fdc_burst =  tau0_fdc_interp[index_mid_burst_t]
+        tau0_fm_rate_burst =  tau0_ka_interp[index_mid_burst_t]
+
+        a0_burst, a1_burst, a2_burst = arr_coeff_a[index_mid_burst_t, :]
+        b0_burst, b1_burst, b2_burst = arr_coeff_b[index_mid_burst_t, :]
+
+        kappa_annotation_burst = (b0_burst
+                                 + b1_burst * (tau_burst - tau0_fm_rate_burst)
+                                 + b2_burst * (tau_burst - tau0_fm_rate_burst)**2)
+
+        kappa_annotation_grid = self._evaluate_polynomial_array(arr_coeff_b,
+                                                                grid_tau,
+                                                                tau0_ka_interp)
+
+        grid_kappa_t = (  (kappa_annotation_grid * kappa_steer_grid)
+                        / (kappa_annotation_grid - kappa_steer_grid))
+        freq_dcg_burst = (  a0_burst
+                          + a1_burst * (tau_burst - tau0_fdc_burst)
+                          + a2_burst * (tau_burst - tau0_fdc_burst)**2)
+
+        grid_freq_dcg = self._evaluate_polynomial_array(arr_coeff_a,
+                                                        grid_tau,
+                                                        tau0_fdc_interp)
+
+        grid_freq_dc = (grid_freq_dcg
+                        + grid_kappa_t * ((grid_t - t_burst)
+                                          + (freq_dcg_burst / kappa_annotation_burst
+                                          - grid_freq_dcg / kappa_annotation_grid)))
+
+        lat_map = gdal.Open(list_filename_llh[0], gdal.GA_ReadOnly).ReadAsArray()
+        lon_map = gdal.Open(list_filename_llh[1], gdal.GA_ReadOnly).ReadAsArray()
+        hgt_map = gdal.Open(list_filename_llh[2], gdal.GA_ReadOnly).ReadAsArray()
```

Comment:
I initially tried the code using `with` as below:

```python
with (gdal.Open(list_filename_llh[0], gdal.GA_ReadOnly) as raster_lat,
      gdal.Open(list_filename_llh[1], gdal.GA_ReadOnly) as raster_lon,
      gdal.Open(list_filename_llh[2], gdal.GA_ReadOnly) as raster_hgt):
    lat_map = raster_lat.ReadAsArray()
    lon_map = raster_lon.ReadAsArray()
    hgt_map = raster_hgt.ReadAsArray()
```

which gave the error below:

```
Traceback (most recent call last):
  File "/Users/jeong/opera-adt/s1-reader_seongsujeong/src/s1reader/tester_s1_reader.py", line 37, in <module>
    azfmrmm = bursts[-1].az_fm_rate_mismatch_mitigation(
  File "/Users/jeong/miniconda3/envs/isce3_dev_rtc/lib/python3.9/site-packages/s1reader/s1_burst_slc.py", line 893, in az_fm_rate_mismatch_mitigation
    with (gdal.Open(list_filename_llh[0], gdal.GA_ReadOnly) as raster_lat,
AttributeError: __enter__
```

Maybe there is some incompatibility for GDAL raster with `with` statement (e.g. [here](https://stackoverflow.com/questions/49648360/use-with-statement-to-close-gdal-datasets) )